src/FormatNumber/Parser.elm

```diff
@@ -378,3 +442,54 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value =
+    let
+        isNegative : Bool
+        isNegative =
+            String.startsWith locale.negativePrefix value
+                && String.endsWith
+                    locale.negativeSuffix
+                    value
+
+        notDigits : Regex.Regex
+        notDigits =
+            "\\D"
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
+        
+        onlyDigits : String -> String
+        onlyDigits = 
+            Regex.replace notDigits (\_ -> "")
+
+        splitValue : String -> List String
```

Comment:
I still think we should use `String -> (String, String)` here.

Reply:
If I change this, then I can move entire logic of `joinParts` into the `parseString`. I like the current structure as it cleanly segregates that out of `parseString` and allow me to test it separately.

Let me know if you still want it changed. I will do it.

Reply:
I think making impossible states impossible worth the job : )